runtime/lua/vim/treesitter/dev.lua

```diff
@@ -438,19 +442,19 @@ function M.inspect_tree(opts)
     group = group,
     buffer = b,
     callback = function()
-      api.nvim_buf_clear_namespace(buf, pg.ns, 0, -1)
+      if api.nvim_buf_is_loaded(buf) then
+        api.nvim_buf_clear_namespace(buf, pg.ns, 0, -1)
+      end
     end,
   })
 
   api.nvim_create_autocmd('BufLeave', {
     group = group,
     buffer = buf,
     callback = function()
-      if not api.nvim_buf_is_loaded(b) then
-        return true
+      if api.nvim_buf_is_loaded(b) then
+        api.nvim_buf_clear_namespace(b, pg.ns, 0, -1)
       end
-
-      api.nvim_buf_clear_namespace(b, pg.ns, 0, -1)
```

Comment:
Similar comment as above. We should keep the `return true` so that the autocommand will delete itself.